test/integration/crud/bulk.test.ts

```diff
@@ -104,6 +104,125 @@ describe('Bulk', function () {
           }
         });
       });
+      context('BulkWriteResult should not include invalid insert in insertedIds', function () {
+        async function insertManyTryInvalidIds(input, isOrdered, indices, expectedInsertedIds) {
+          try {
+            const db = client.db();
+            const col = db.collection('test');
+            for (let i = 0; i < indices.length; i++) {
+              await col.createIndex(indices[i], { unique: true, sparse: false });
+            }
+            await col.insertMany(input, { ordered: isOrdered });
+            expect(false); // no error -> test fails
```

Comment:
```suggestion
            expect.fail('a MongoBulkWriteError must be thrown when write errors are encountered');
```

Reply:
It makes sense to have a more specific error statement here.